### DISCUSSION TITLE
v0.11.x

### DISCUSSION THREAD

**Discussion by gerwaric** [gerwaric/acquisition#37] [Announcements]
## The first release with OAuth support

- This update brings OAuth support, but POESESSID should still work.
- This release also brings back macOS and Linux support.

*IMPORTANT:* While working on this release, I realized that acquisition has not been properly calculating pseudo mods. I think this may have been going on for a while. This means, for example, that if an item has two maximum life modifiers or two modifiers that provide chaos resistance, they will not be added together. I've already created https://github.com/gerwaric/acquisition/issues/31 for this issue.

Please report any other issues here: https://github.com/gerwaric/acquisition/issues

### OAuth support has arrived
- You can now login either with OAuth as using the traditional POESESSID
- Right now Acquisition is what's called a public OAuth client. The other option is a confidential client, which is more secure and doesn't require you to re-authenticate as often. Upgrading acquisition to a confidential client is on the roadmap.
- You can see the details of your OAuth token with new options in the Settings menu.
- You still need a POESESSID cookie to post or update forum shop threads. There is an entry in the Settings menu where you can do this.

### Update-checking has changed
- Acquisition now requests releases directly from GitHub instead of checking versions.txt. This allows acquisition to show you exactly what new version are available.
- The update dialog box give you the choice to open the github releases page, or ignore the update and continue, or ignore the update until new versions are available.

### UI changes
- The status bar updates should be a little more informative
- The rate limit status panel has been replaced by a dialog box

### Bug fixes
- Acquisition no longer blocks windows from shutting down
- Fixed a couple bugs that were causing errors or crashes when refreshing items 
- Some edge cases involving the rate limiter have been fixed

### Under the hood changes and improvements
- Implemented an embarrassingly basic form of dependency injection 
- Some settings have been moved from the data store into the settings.ini
- Classes defined in the some of the more sprawling files have been refactored into their own files
- Simplified or removed signaling between certain objects
- The rate limiter is only used for rate-limited requests now
- Removed old code that's no longer needed, e.g. porting.h and the self-limiting reply

## What's yet to come

- Making acquisition a confidential OAuth client for better authentication support
- Adding support for pseudo-mods
- Updating Boost and other submodules as needed (maybe replacing them with subtrees)
- More internal refactoring to make the guts simpler and hopefully easier to understand

<hr /><em>This discussion was created from the release <a href='https://github.com/gerwaric/acquisition/releases/tag/v0.11.0'>v0.11.0</a>.</em>

**Comment by jhc1993-846** [comment]
I just wanted to thank you for making this work again, on _Linux_ too no less!
This will certainly make my Path of Exile much easier.

**Reply by gerwaric** [reply]
Thank you very much. It's been a massive learning experience, and I'm glad to hear it's useful!
